internal/cow/cow.go

```diff
@@ -11,39 +12,39 @@ type Process interface {
 	// Close releases resources associated with the process and closes the
 	// writer and readers returned by Stdio. Depending on the implementation,
 	// this may also terminate the process.
-	Close() error
+	Close(ctx context.Context) error
```

Comment:
I don't like that this takes a context either. This is about releasing resources; it shouldn't require an RPC to the guest.

The error should still get wrapped and propagated up appropriately, so we shouldn't need to log inside. I think it only makes sense to pass a ctx when we're calling out to the guest so that there is invisible state that can't be logged externally.

Reply:
Not sure I have a strong argument for `Close`. In general its nice to have the logs show the context associated with the caller of `Close`. Just helps to see when the lifetime model is being called. But I don't care that much and can remove for `Close.